swdb_2018_tools/ephys/get_stimulus_blocks.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+"""
+Created on Mon Aug 27
+
+@author: shawno
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+def get_stimulus_blocks(data_set):
+    '''Return dataframe with start and end times for each stimulus block '''
+    stim_blocks = []
+    for i,k in enumerate(data_set.stim_tables.keys()):
+
+        stim_table = data_set.stim_tables[k]
+        s = stim_table.start.values
+        s_end = stim_table.end.values
+        idx = np.where(np.diff(s)>100)[0]
+
```

Comment:
This could fail silently if there are more than 3 blocks. Maybe something like this:

```
idx = np.where(np.diff(s)>100)[0] + 1
full_idx = np.sort(np.unique(np.concatenate([idx, [0, len(s)]])))

stim_blocks = []

for low, high in zip(full_idx[:-1], full_idx[1:]):
    stim_blocks.append((k, s[low], s_end[high-1]))
```

Reply:
`100` could also be a default parameter value.